### PR TITLE
Chem how to doc

### PR DESCRIPTION
**Pull request by aladshaw3**
## Fixes/Addresses:

#113  Basic how-to on chemistry 

## Summary/Motivation:

PR addresses need for a how-to guide for basic chemistry using the generic IDAES framework. 

## Changes proposed in this PR:
- Added a how-to on chemistry 
- Modified index file to point to chemistry how-to

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/how_to_guides/how_to_setup_simple_chemistry.rst
Comment (adam-a-a) [inline docs/how_to_guides/how_to_setup_simple_chemistry.rst:230]:
```suggestion
    ProteusLib (Documentation pending)

```

File: docs/how_to_guides/how_to_setup_simple_chemistry.rst
Comment (adam-a-a) [inline docs/how_to_guides/how_to_setup_simple_chemistry.rst:310]:
```suggestion
                       #    thus, this reaction is of order 0 with respect

```

File: docs/how_to_guides/how_to_setup_simple_chemistry.rst
Comment (adam-a-a) [inline docs/how_to_guides/how_to_setup_simple_chemistry.rst:405]:
```suggestion
    from idaes.generic_models.properties.core.generic.generic_reaction import GenericReactionParameterBlock

```

File: docs/how_to_guides/how_to_setup_simple_chemistry.rst
Comment (adam-a-a) [inline docs/how_to_guides/how_to_setup_simple_chemistry.rst:96]: I think there may be an error somewhere in `thermo_config` because when I try running the code blocks, I get an error. I tried fixing by adding a missing curly brace but then get another error. We may want to incorporate testing like @bknueven did for previous code blocks (although maybe we can get that through in another PR).

File: docs/how_to_guides/how_to_setup_simple_chemistry.rst
Comment (adam-a-a) [inline docs/how_to_guides/how_to_setup_simple_chemistry.rst:133]: I think adding a curly brace here fixes the other error.
```suggestion
                        }},

```

**Review by adam-a-a — CHANGES_REQUESTED** [review]
I outlined a few minor changes and a couple of bug fixes in code blocks. I think once those changes are made, we can merge this--unless we want to add testing now rather than wait for a subsequent PR.

Also, I enjoyed going through this how-to. I thought the explanations were thorough and clear, and the doc helped me understand how to setup chemistry for a unit model-- so it does what the title claims.  I can foresee the need to break this down into separate chunks, if we are to apply the Diataxis framework later on.

**Comment by bknueven** [comment]
I think doc-testing could be saved for a follow-on PR. Given the structure I think it's easy to add-on to what's here -- it doesn't look like it would require re-working the examples.

**Comment by adam-a-a** [comment]
I added testing and fixed the minor bugs in the code blocks. Adding @bknueven as a reviewer so that we can push this through

**Review by adam-a-a — APPROVED** [review]

**Review by bknueven — APPROVED** [review]
The inline code is tested to ensure it executes and the documentation seems reasonable with nice references into the IDAES documentation. I wonder if there's an easy way to check for broken links automatically, since that will probably burn us at some point.

**Comment by aladshaw3** [comment]
> 
> 
> The inline code is tested to ensure it executes and the documentation seems reasonable with nice references into the IDAES documentation. I wonder if there's an easy way to check for broken links automatically, since that will probably burn us at some point.

This would be a good thing to dig into. The chemistry modules are heavily dependent on the core IDAES and we will need to constantly refer back to IDAES documentation in order to avoid duplication of documentation for specific sub-modules.

**Pull request merged**
